src/components/plugins/search/SearchBar.jsx

```diff
@@ -75,6 +75,14 @@ const SearchBar = ({
 
   const tagItems = tagsList.map((item) => `${item.dataIndex}_${item.text}`);
 
+  const staleTags = tagItems.filter(
+    (tagItem) => !listItems.find((listItem) => listItem.value === tagItem),
+  );
+  if (staleTags.length > 0) {
+    staleTags.forEach((staleTag) => searchHandler.removeTag(staleTag));
+    updateGridState();
+  }
+
```

Comment:
The component should be broken into smaller functions for maintainability and code readability. [Will be added to next pull request]